src/Layout.js

```diff
@@ -0,0 +1,45 @@
+import {useState, useEffect} from 'react'
+import {ThemeContext, themeContextDefault} from './contexts/theme.js'
+
+export default function Layout({children}) {
+	const toggleTheme = () => {
+		const {themes, theme} = themeContext
+		const themesLength = themes.length
+		const activeThemeIndex = themes.indexOf(theme)
+		const newThemeIndex = themesLength % activeThemeIndex
+		const newTheme = themes[newThemeIndex]
+		setTheme(newTheme)
+	}
+
+	const [theme, setTheme] = useState(themeContextDefault.theme)
+
+	const themeContext = {
+		theme,
+		themes: themeContextDefault.themes,
+		userBrowserTheme: themeContextDefault.userBrowserTheme,
+		toggleTheme: toggleTheme
+	}
+
+	const darkModeListener = (event) => {
+		if (event.matches) {
+			setTheme(themeContextDefault.themes[0])
+		} else {
+			setTheme(themeContextDefault.themes[1])
+		}
+	}
+
+	useEffect(() => {
+		window.addEventListener('change', darkModeListener)
```

Comment:
not sure why this is never called

Reply:
the evenlistener should not be on `window` but `window.matchMedia`. Like this:

https://github.com/internet4000/radio4000-cms/blob/a31ca31579e90b2bb3a235cfefe72cb7322fb388/src/contexts/theme.js#L13